geoips/plugins/modules/output_checkers/text_checker.py

```diff
@@ -0,0 +1,122 @@
+            line = f.readline()
+        if isinstance(line, str):
+            return True
+    return False
+
+
+def outputs_match(plugin, output_product, compare_product):
+    """Check if two text files match.
+
+    Parameters
+    ----------
+    plugin: OutputCheckerPlugin
+        The corresponding geotiff OutputCheckerPlugin that has access to needed methods
+    output_product : str
+        Full path to current output product
+    compare_product : str
+        Full path to "good" comparison product
+
+    Returns
+    -------
+    bool
+        Return True if products match, False if they differ
+    """
+    ret = subprocess.run(
+        ["diff", output_product, compare_product],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    LOG.debug(ret.stdout)
+    if ret.returncode == 0:
+        LOG.info("    *****************************")
+        LOG.info("    *** GOOD Text files match ***")
+        LOG.info("    *****************************")
+        return True
+
+    out_difftxt = plugin.get_out_diff_fname(compare_product, output_product)
+    with open(out_difftxt, "w") as fobj:
+        subprocess.call(["diff", output_product, compare_product], stdout=fobj)
+    LOG.interactive("    *******************************************")
+    LOG.interactive("    *** BAD Text files do NOT match exactly ***")
+    LOG.interactive("    ***   output_product: %s ***", output_product)
+    LOG.interactive("    ***   compare_product: %s ***", compare_product)
+    LOG.interactive("    ***   out_difftxt: %s ***", out_difftxt)
+    LOG.interactive("    *******************************************")
+    return False
+
+
+def call(plugin, compare_path, output_products, test_product_func=None):
+    """Compare the "correct" text found the list of current output_products.
+
+    Compares files produced in the current processing run with the list of
+    "correct" files contained in "compare_path".
+
+    Parameters
+    ----------
+    plugin: OutputCheckerPlugin
+        The corresponding text OutputCheckerPlugin that has access to needed methods
+    compare_path : str
+        Path to directory of "correct" products - filenames must match output_products
+    output_products : list of str
+        List of strings of current output products,
+        to compare with products in compare_path
+    test_product_func : function, default=None
+        Alternative function to be used for testing output product
+
+          * Call signature must be:
+
+              * output_product, compare_product, goodcomps, badcomps, compare_strings
+
+          * Return must be:
+
+              * goodcomps, badcomps, compare_strings
+
+        * If None, use geoips.compare_outputs.test_product)
+
+    Returns
+    -------
+    int
+        Binary code: 0 if all comparisons were completed successfully.
+    """
+    retval = plugin.compare_outputs(compare_path, output_products, test_product_func)
+    return retval
```

Comment:
Can we remove the need for "test_product_func" ?  Shouldn't it be assumed now that we have separate output checker plugins that the "test_product_func" is just whatever is defined internally to the current output checker?

Do we just need to specify additional required methods for this interface that determine how to compare the outputs, and how to check if a given output is of the type required to use this output checker?